internal/expr/bindtypes.go

```diff
@@ -45,7 +45,10 @@ func (pe *ParsedExpr) BindTypes(args ...any) (tbe *TypeBoundExpr, err error) {
 
 	// Bind types to each expression.
 	var typedExprs TypeBoundExpr
-	outputUsed := map[typeinfo.Output]bool{}
+	// outputUsed records if the string representation of an output appears
+	// more than once in the query. If the string appears more than once then
+	// there is ambiguity in the query.
```

Comment:
I think now this comment is a little bit superfluous because it is merely stating that the String representation is unique, which seems to be the case now that we have a specific function.